File: baekjoon_all/10000+/boj_10845_solution2.py
# Solve 2022-06-01
# Update 2023-03-01

import sys

input = lambda : sys.stdin.readline().rstrip()

n = int(input())
q = []

for _ in range(n):
    line = input().split()

    if line[0] == "push":
        q.insert(0, line[1])
    elif line[0] == "pop":
        if len(q) == 0:
            print(-1)
        else:
            print(q.pop())
    elif line[0] == "size":
        print(len(q))
    elif line[0] == "empty":
        if len(q) == 0:
            print(1)
        else:
            print(0)
    elif line[0] == "front":
        if len(q) == 0:
            print(-1)
        else:
            print(q[-1])
    else:
        if len(q) == 0:
            print(-1)
        else:
            print(q[0])

File: baekjoon_all/10000+/boj_10845_solution2.cpp
// Solve 2022-06-05
// Update 2023-03-01

#include <bits/stdc++.h>
using namespace std;

#ifdef BOJ
#define BOJTEST(x) ((void)0)
#else
#define BOJTEST(x) cout << "[Debug] " << #x << ':' << x << '\n'
#endif
#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL); // boj_15552.cpp
#define SETPRECISION(n) cout << fixed;cout.precision(n); // boj_1008.cpp
#define SIZE(v) (int)v.size()
#define ALL(v) v.begin(),v.end()
using ll = long long;
using uint = unsigned int;
using ull = unsigned long long;

int q_begin = 0, q_end = 0;
int q[10001];

void push(int x) {
    q[q_end] = x;
    q_end++;
}

int pop() {
    if (q_begin == q_end) {
        return -1;
    }
    q_begin++;
    return q[q_begin - 1];
}

int size() {
    return q_end - q_begin;
}

int empty() {
    if (q_begin == q_end) {
        return 1;
    }
    return 0;
}

int front() {
    if (q_begin == q_end) {
        return -1;
    }
    return q[q_begin];
}

int back() {
    if (q_begin == q_end) {
        return -1;
    }
    return q[q_end - 1];
}

int main() {
    FASTIO;

    int n;
    cin >> n;

    for (int i = 0; i < n; i++) {
        string cmd;
        cin >> cmd;

        if (cmd == "push") {
            int x;
            cin >> x;
            push(x);
        }
        else if (cmd == "pop") {
            cout << pop() << '\n';
        }
        else if (cmd == "size") {
            cout << size() << '\n';
        }
        else if (cmd == "empty") {
            cout << empty() << '\n';
        }
        else if (cmd == "front") {
            cout << front() << '\n';
        }
        else {
            cout << back() << '\n';
        }
    }

    return 0;
}